great_expectations/checkpoint/util.py

```diff
@@ -245,9 +245,11 @@ def get_substituted_batch_request(
     for key, value in validation_batch_request.items():
         substituted_value = substituted_runtime_batch_request.get(key)
         if value is not None and substituted_value is not None:
-            raise ge_exceptions.CheckpointError(
-                f'BatchRequest attribute "{key}" was specified in both validation and top-level CheckpointConfig.'
-            )
+            validation_batch_request[key] = substituted_value
+            substituted_runtime_batch_request.pop(key)
+            # raise ge_exceptions.CheckpointError(
+            #     f'BatchRequest attribute "{key}" was specified in both validation and top-level CheckpointConfig.'
+            # )
```

Comment:
This was to address a conflict between `mercury`,  how GX does final substitutions to yield a complete validation config at runtime, and the signature of `RuntimeBatchRequest`. Exception got thrown when attempting to run the checkpoint using a runtime datasource: 

```
batch_request = RuntimeBatchRequest(
            data_connector_name="default_runtime_data_connector",
            datasource_name="Residential Data",
            data_asset_name="raw-residential-data",
            runtime_parameters={
                "batch_data": df_success
            },
            batch_identifiers={
                "color": "blue"
            }
        )

        context.run_checkpoint(
            ge_cloud_id=checkpoint_id,
            batch_request=batch_request,
            run_name=f"{i}-run-run-run"
        )
```

Quoted from[ comment in different PR](https://github.com/superconductive/mercury/pull/1083#discussion_r954371365):

" There's a bit of a conflict here - in mercury, if a batch_request appears in validations, it has to have complete datasource_name/data_asset_name/data_connector_name triple. Likewise, datasource_name and data_connector_name are required positional args for RuntimeBatchRequest.

BUT, when you satisfy those two constraints, it breaks the logic surrounding checkpoint config substitution at runtime WRT to batch requests - it can only be an exclusive union (is this the right term?). An attr can't appear in both a batch request passed in at runtime and in batch_request at top-level, or within validations.

My instinct is that we should probably review some of that update precedence logic in OSS GX and also make sure there aren't other potential places where mercury conflicts. cc: @cdkini @Kilo59"

Reply:
I don't think substituting is the right thing to do here. The runtime batch request is really not a batch request, it's just the runtime params + batch identifiers. The workaround is to not use a `RuntimeBatchRequest` object here

```python
batch_request = {
    "runtime_parameters": {
        "batch_data": df_success
    },
    "batch_identifiers": {
        "color": "blue"
    },
}
checkpoint.run(batch_request=batch_request)
```